libs/code-demos/src/lib/multitab-editor/multitab-editor.component.ts

```diff
@@ -147,8 +168,53 @@ export class MultitabEditorComponent
     this.dispose();
   }
 
-  trackByEditorIndex(index, model) {
-    return model.editorIndex;
+  trackByEditorIndex(index: number, model: MonacoModel) {
+    return model.path;
+  }
+
+  getFileOnlyFromPath(path: string): string {
+    const pathFrags = path.split('/');
+    return pathFrags.length > 0 ? pathFrags.pop() : path;
+  }
+
+  getActiveModel(pathName: string) {
+    return (this.models.find(m => m.path === pathName) || {model: null, highlight: null});
+  }
+
+  onSelectedTabChange({ index }) {
+    const model = this.openModels[index];
+    if (model) {
+      this.updateActiveFileSelected(model);
+    }
+  }
+
+
+  updateActiveFileSelected(model: FileFolder|MonacoModel) {
+    const alreadyShownInTab = this.openModels.findIndex(v => v.path === model.path);
+
+    if (alreadyShownInTab < 0) {
+      const monacoModel = this.models.find(m => m.path === model.path);
+      this.openModels = [...this.openModels, monacoModel];
+      this.activeTabIndex = this.openModels.length - 1;
+    } else {
+      this.activeTabIndex = alreadyShownInTab;
+    }
+  }
+
+
+  onCloseTab(model: FileFolder) {
```

Comment:
Would it be hard to have tabs and file nave in a separate components?

Reply:
no that was actually my first thought which is why i opted to use templateOutlets to abstract them out - wasn't sure if you wanted each part of the multitab component as their own components as well. 

I'll move them to their own component.

Reply:
I'll do this in another PR as it will be a lot of changes. Planning to also update the styles to sass so we can clean up verbosity and start creating variables for consistency.